.teamcity/_Self/templates/TestTemplateBase.kt

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+package _self.templates
+
+import jetbrains.buildServer.configs.kotlin.v2019_2.*
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.freeDiskSpace
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.swabra
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.script
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.vcs
+
+open class TestTemplateBase : Template({
+    name = "Python Test Template Base Class"
+    id = RelativeId("TestTemplateBase")
+
+    artifactRules = "+:%project.directory%/htmlcov => coverage.zip"
```

Comment:
The Python coverage project outputs coverage results in formats that don't work directly in TeamCity. There's probably a plugin somewhere that supports the Java Cobertura code coverage format used by the python coverage XML reporter... but I didn't find it and it isn't important enough. I want to be able to see the code coverage, but I'm not going to fail a build because of code coverage. We don't do that in our other Ed-Fi projects. Therefore an HTML report is appropriate. Here in this artifact rule we simply create a zip file of the `htmlcov` directory that is created by the coverage utility.